prometheus/promhttp/http.go

```diff
@@ -90,6 +93,13 @@ func HandlerFor(reg prometheus.Gatherer, opts HandlerOpts) http.Handler {
 	return HandlerForTransactional(prometheus.ToTransactionalGatherer(reg), opts)
 }
 
+// HandlerWithAllowlist remove metrics whose name can't match with metricNameMatcher
+func HandlerWithAllowlist(metricNameMatcher *regexp.Regexp) http.Handler {
```

Comment:
This is nice, but we might want to discuss exact API. e.g. do we want only metric names or also labels? What would be flexible enough?

We think it might be useful to tackle that with OpenMetrics changes and have same API for per scrape or per handler options 🤔